src/auth_code.rs

```diff
@@ -85,18 +87,18 @@ impl BaseClient for AuthCodeSpotify {
         &self.config
     }
 
-    async fn get_token(&self) -> Ref<Option<Token>> {
+    async fn get_token(&self) -> RwLockReadGuard<Option<Token>> {
         self.auto_reauth()
             .await
             .expect("Failed to re-authenticate automatically, please authenticate");
-        self.token.borrow()
+        self.token.read().unwrap()
```

Comment:
Perhaps we could handle the exception more elegantly than `unwrap`?

Reply:
sure!